server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/ValidRcptHandlerTest.java

```diff
@@ -103,63 +111,96 @@ public Object getAttachment(String key, State state) {
+        MailAddress mailAddress = new MailAddress(INVALID_USER + "@otherdomain");
+        SMTPSession session = setupMockedSMTPSession(!RELAYING_ALLOWED);
+
+        int rCode = handler.doRcpt(session, SENDER, mailAddress).getResult();
+
+        assertThat(rCode).isEqualTo(HookReturnCode.DECLINED                         );
     }
 
     @Test
-    public void testNotRejectValidUser() throws Exception {
-        MailAddress mailAddress = new MailAddress(VALID_USER + "@localhost");
-        SMTPSession session = setupMockedSMTPSession(false);
+    public void doRcptShouldDeclineValidUsersWhenNoRelay() throws Exception {
+        SMTPSession session = setupMockedSMTPSession(!RELAYING_ALLOWED);
 
-        int rCode = handler.doRcpt(session, null, mailAddress).getResult();
+        int rCode = handler.doRcpt(session, SENDER, validUserEmail).getResult();
 
         assertThat(rCode).isEqualTo(HookReturnCode.DECLINED);
     }
 
     @Test
-    public void testHasAddressMapping() throws Exception {
-        MailAddress mailAddress = new MailAddress(USER1 + "@localhost");
-        SMTPSession session = setupMockedSMTPSession(false);
+    public void doRcptShouldDeclineValidUsersWhenRelay() throws Exception {
+        SMTPSession session = setupMockedSMTPSession(RELAYING_ALLOWED);
 
-        int rCode = handler.doRcpt(session, null, mailAddress).getResult();
+        int rCode = handler.doRcpt(session, SENDER, validUserEmail).getResult();
 
         assertThat(rCode).isEqualTo(HookReturnCode.DECLINED);
     }
 
     @Test
-    public void testHasErrorMapping() throws Exception {
-        MailAddress mailAddress = new MailAddress(USER2 + "@localhost");
-        SMTPSession session = setupMockedSMTPSession(false);
+    public void doRcptShouldDeclineWhenHasAddressMapping() throws Exception {
+        memoryRecipientRewriteTable.addAddressMapping(USER1, Domain.LOCALHOST, "address");
 
-        int rCode = handler.doRcpt(session, null, mailAddress).getResult();
+        SMTPSession session = setupMockedSMTPSession(!RELAYING_ALLOWED);
 
-        assertThat(session.getAttachment("VALID_USER", State.Transaction)).isNull();
-        assertThat(rCode).isEqualTo(HookReturnCode.DENY);
+        int rCode = handler.doRcpt(session, SENDER, validUserEmail).getResult();
+
+        assertThat(rCode).isEqualTo(HookReturnCode.DECLINED);
+    }
+
+    @Ignore
+    @Test
+    public void doRcptShouldDenyWhenHasMappingLoop() throws Exception {
+        memoryRecipientRewriteTable.addAddressMapping(USER1, Domain.LOCALHOST, USER2 + "@localhost");
+        memoryRecipientRewriteTable.addAddressMapping(USER2, Domain.LOCALHOST, USER1 + "@localhost");
+
+        SMTPSession session = setupMockedSMTPSession(!RELAYING_ALLOWED);
+
+        int rCode = handler.doRcpt(session, SENDER, user1mail).getResult();
+
+        assertThat(rCode).isEqualTo(HookReturnCode.DECLINED);
+    }
+
+    @Ignore
+    @Test
+    public void doRcptShouldDenyWhenHasErrorMapping() throws Exception {
```

Comment:
it seems to be declined

Reply:
Looks like a rebase fail. (I altered the behaviour of that method then I reverted it (cf discussion about log message above).